server/client.go

```diff
@@ -3611,10 +3611,12 @@ func (c *client) processMsgResults(acc *Account, r *SublistResult, msg, deliver,
 			ql := _ql[:0]
 			for i := 0; i < len(qsubs); i++ {
 				sub = qsubs[i]
-				if sub.client.kind == CLIENT {
+				if sub.client.kind == LEAF || sub.client.kind == ROUTER || sub.client.kind == GATEWAY {
```

Comment:
Since for gateways interest is not stored in the account sublist (but stored in gateway struct), sub.client.kind will never be GATEWAY.